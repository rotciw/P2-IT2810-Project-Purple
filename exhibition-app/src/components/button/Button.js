import React from 'react'
import "./Button.css"

export default function Button(props){
    let className = "category_button";

    if (!isNaN(props.name)){
        className += " alternative_button";
    }

    if (props.selected_category === props.id) {
        className += " selected_category_button"
    }


    return (
            <button className={className} onClick={props.onClick}>
            {props.name}
            </button>
    )
}
