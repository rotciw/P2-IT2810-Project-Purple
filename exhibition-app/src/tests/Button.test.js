import React from 'react';
import Button from '../components/button/Button.js';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Button className="test" onClick="test.onClick">test</Button>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
