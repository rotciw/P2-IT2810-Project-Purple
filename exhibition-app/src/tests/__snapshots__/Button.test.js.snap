// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="category_button selected_category_button"
  onClick="test.onClick"
/>
`;
